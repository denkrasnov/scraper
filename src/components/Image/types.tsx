export type ObjectFit = "contain" | "cover" | "fill" | "scale-down";

export interface ImageProps {
  alt: string;
  src: string;
  objectFit?: ObjectFit;
}
