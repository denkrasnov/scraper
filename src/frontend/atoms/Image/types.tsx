export type ObjectFit = "cover" | "fill" | "scale-down";

export interface ImageProps {
  alt: string;
  src: string;
  objectFit?: ObjectFit;
  borderRadiusLeft?: boolean;
  width?: string;
  height?: string;
}
