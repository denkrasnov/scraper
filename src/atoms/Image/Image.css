.Image {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.borderRadiusLeft {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}
