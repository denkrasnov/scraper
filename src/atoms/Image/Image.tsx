import React, { FC } from "react";

import { ImageProps } from "./types";
import styles from "./Image.css";

const Image: FC<ImageProps> = props => {
  const { alt, borderRadiusLeft, objectFit, src } = props;

  return (
    <img
      alt={alt}
      className={borderRadiusLeft ? styles.borderRadiusLeft : styles.Image}
      src={src}
      style={{ objectFit }}
      width="100%"
    />
  );
};

export default Image;
