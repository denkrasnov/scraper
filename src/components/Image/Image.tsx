import React, { FC } from "react";

import { ImageProps } from "./types";

const Image: FC<ImageProps> = props => {
  const { alt, objectFit, src } = props;

  return <img alt={alt} src={src} style={{ objectFit }} />;
};

export default Image;
