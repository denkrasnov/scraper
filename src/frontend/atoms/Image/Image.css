.Image {
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  object-fit: contain;
}

.borderRadiusLeft {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}
