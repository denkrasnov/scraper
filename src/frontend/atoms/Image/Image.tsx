import { FC } from "react";

import mapPropsToStyles from "../helpers/mapPropsToStyles";
import { ImageProps } from "./types";
import styles from "./Image.css";

const Image: FC<ImageProps> = (props) => {
  const { alt, src, width, height, ...restProps } = props;

  const classes = mapPropsToStyles(restProps, styles);
  return (
    <img alt={alt} className={classes} src={src} style={{ width, height }} />
  );
};

export default Image;
