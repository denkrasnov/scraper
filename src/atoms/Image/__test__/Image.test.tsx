import React from "react";
import { shallow } from "enzyme";

import Image from "..";

describe("Button", () => {
  it("renders", () => {
    const component = shallow(<Image alt="__ALT__" src="__SRC__" />);
    expect(component).toMatchSnapshot();
  });

  it("renders with borderRadiusLeft", () => {
    const component = shallow(
      <Image alt="__ALT__" borderRadiusLeft src="__SRC__" />
    );
    expect(component).toMatchSnapshot();
  });
});
