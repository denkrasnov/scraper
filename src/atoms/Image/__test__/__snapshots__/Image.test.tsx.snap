// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<img
  alt="__ALT__"
  className="Image"
  src="__SRC__"
  style={
    Object {
      "objectFit": undefined,
    }
  }
  width="100%"
/>
`;

exports[`Button renders with borderRadiusLeft 1`] = `
<img
  alt="__ALT__"
  className="borderRadiusLeft"
  src="__SRC__"
  style={
    Object {
      "objectFit": undefined,
    }
  }
  width="100%"
/>
`;
